Add autoload support for different casual menus

kickingvegas opened the pull request:
- The following menus are now autoloaded:
  - dired
    - casual-dired-sort-by-tmenu
    - casual-dired-search-replace-tmenu
  - editkit
    - casual-editkit-open-tmenu
    - casual-editkit-project-tmenu
    - casual-editkit-edit-tmenu
    - casual-editkit-emoji-symbols-tmenu
    - casual-editkit-mark-tmenu
    - casual-editkit-copy-tmenu
    - casual-editkit-kill-tmenu
    - casual-editkit-sort-tmenu
    - casual-editkit-transpose-tmenu
    - casual-editkit-delete-tmenu
    - casual-editkit-move-text-tmenu
    - casual-editkit-windows-tmenu
    - casual-editkit-windows-delete-tmenu
    - casual-editkit-bookmarks-tmenu
    - casual-editkit-search-tmenu
    - casual-editkit-tools-tmenu
    - casual-editkit-registers-tmenu
    - casual-editkit-rectangle-tmenu
    - casual-editkit-transform-text-tmenu
    - casual-editkit-macro-tmenu
  - ibuffer
    - casual-ibuffer-filter-tmenu
    - casual-ibuffer-sortby-tmenu
  - isearch
    - casual-isearch-tmenu

- Fix unit test.

swflint commented:
There are some features not supported, because they use the hyperbolic flag.  I'm not quite sure how best to add them in at this time.

kickingvegas commented:
Thank you so much for your interest and initiative in submitting a PR to Casual! That said there is preliminary planning and thinking in-progress on supporting algebra/symbolic functions in Casual that is milestoned for the 1.5.0 release. This work is unfortunately incompatible with your submission and as such, is rejected.

That said, if you wish to contribute in this regard, I'm open for having a Zoom call where I can share my thoughts on methodology and conventions for developing Casual. If you're comfortable with this, then we can move this conversation forward.

All my best -
Charles

kickingvegas committed:
Add autoload support for different casual menus

- The following menus are now autoloaded:
  - dired
    - casual-dired-sort-by-tmenu
    - casual-dired-search-replace-tmenu
  - editkit
    - casual-editkit-open-tmenu
    - casual-editkit-project-tmenu
    - casual-editkit-edit-tmenu
    - casual-editkit-emoji-symbols-tmenu
    - casual-editkit-mark-tmenu
    - casual-editkit-copy-tmenu
    - casual-editkit-kill-tmenu
    - casual-editkit-sort-tmenu
    - casual-editkit-transpose-tmenu
    - casual-editkit-delete-tmenu
    - casual-editkit-move-text-tmenu
    - casual-editkit-windows-tmenu
    - casual-editkit-windows-delete-tmenu
    - casual-editkit-bookmarks-tmenu
    - casual-editkit-search-tmenu
    - casual-editkit-tools-tmenu
    - casual-editkit-registers-tmenu
    - casual-editkit-rectangle-tmenu
    - casual-editkit-transform-text-tmenu
    - casual-editkit-macro-tmenu
  - ibuffer
    - casual-ibuffer-filter-tmenu
    - casual-ibuffer-sortby-tmenu
  - isearch
    - casual-isearch-tmenu

- Fix unit test.
